core/common/src/main/java/alluxio/underfs/BaseUnderFileSystem.java

```diff
@@ -134,10 +123,11 @@ public AlluxioURI resolveUri(AlluxioURI ufsBaseUri, String alluxioPath) {
         PathUtils.concatPath(ufsBaseUri.getPath(), alluxioPath), ufsBaseUri.getQueryMap());
   }
 
-  @Override
-  public void setProperties(Map<String, String> properties) {
-    mProperties.clear();
-    mProperties.putAll(properties);
+  /**
+   * @return whether only read operations are permitted to the {@link UnderFileSystem}
+   */
+  protected boolean isReadOnly() {
+    return mUfsConf.isReadOnly();
```

Comment:
This doesn't seem scalable since we would have a new method for each configuration value, could we do something like `getUfsConfiguration().get(PropertyKey.UFS_CONF_IS_READ_ONLY)`?